exasol_advanced_analytics_framework/udf_communication/peer_communicator/background_listener_interface.py

```diff
@@ -48,22 +49,22 @@ def __init__(self,
         self._thread.start()
         self._set_my_connection_info()
 
-    def _create_in_control_socket(self, context) -> str:
-        self._in_control_socket: zmq.Socket = context.socket(zmq.PAIR)
+    def _create_in_control_socket(self, socket_factory: SocketFactory) -> str:
+        self._in_control_socket: Socket = socket_factory.create_socket(SocketType.PAIR)
         in_control_socket_address = f"inproc://BackgroundListener_in_control_socket{id(self)}"
         self._in_control_socket.bind(in_control_socket_address)
         return in_control_socket_address
 
-    def _create_out_control_socket(self, context) -> str:
-        self._out_control_socket: zmq.Socket = context.socket(zmq.PAIR)
+    def _create_out_control_socket(self, socket_factory: SocketFactory) -> str:
+        self._out_control_socket: Socket = socket_factory.create_socket(SocketType.PAIR)
         out_control_socket_address = f"inproc://BackgroundListener_out_control_socket{id(self)}"
         self._out_control_socket.bind(out_control_socket_address)
         return out_control_socket_address
```

Comment:
:spiral_notepad:  Notes:
* Could me merged into a single function, if direction gets parameterized.
* Returning socket and socket address and doing the assignment(s) at the call site, may make it easier to see/spot the side effects

Reply:
This I need to check in depth.